builder/amazon/common/ami_config.go

```diff
@@ -20,14 +20,17 @@ type AMIConfig struct {
 	AMIEnhancedNetworking   bool              `mapstructure:"enhanced_networking"`
 	AMIForceDeregister      bool              `mapstructure:"force_deregister"`
 	AMIEncryptBootVolume    bool              `mapstructure:"encrypt_boot"`
+	AMISkipRegister         bool              `mapstructure:"skip_register_ami"`
 }
 
 func (c *AMIConfig) Prepare(ctx *interpolate.Context) []error {
 	var errs []error
-	if c.AMIName == "" {
+	if c.AMIName == "" && !c.AMISkipRegister {
```

Comment:
I think I was referring to `&awscommon.StepPreValidate{`. I would add another struct element to it, `SkipRegister`, and inside the step, check if that flag is set and return early if it is